telethon/helpers.py

```diff
@@ -95,17 +95,20 @@ def strip_text(text, entities):
     return text
 
 
-def retry_range(retries):
+def retry_range(retries, reconnect=False):
```

Comment:
This way it will try to re_connect 1 time even if the retries are set to 0.
But using the special parameter will ensure that the reconnect is not initiated. 
We can assign None if retries are 0 above the loop but this will contradict the below I guess?
The logic will be inside the helper func as lonami said.